File: Tablut/src/it/unibo/ai/didattica/competition/tablut/exceptions/OccupiedException.java
package it.unibo.ai.didattica.competition.tablut.exceptions;

import it.unibo.ai.didattica.competition.tablut.domain.Action;

import java.io.Serial;

/**
 * This exception represent an action that is moving to an occupited box
 * @author A.Piretti
 *
 */
public class OccupiedException extends Exception {

@Serial
private static final long serialVersionUID = 1L;
	
	public OccupiedException(Action a)
	{
		super("Move into a box occupited form another pawn: "+a.toString());
	}

	
}

File: Tablut/src/it/unibo/ai/didattica/competition/tablut/exceptions/OccupiedException.kt
package it.unibo.ai.didattica.competition.tablut.exceptions

import it.unibo.ai.didattica.competition.tablut.domain.Action
import java.io.Serial

/**
 * This exception represent an action that is moving to an occupited box
 * @author A.Piretti
 */
class OccupiedException(a: Action) : Exception("Move into a box occupited form another pawn: $a") {
    companion object {
        @Serial
        private val serialVersionUID = 1L
    }
}